packages/components/Menu/src/MenuItemRadio/MenuItemRadio.android.tsx

```diff
@@ -0,0 +1,71 @@
+/** @jsx withSlots */
+import React from 'react';
+import { Image, Pressable, View } from 'react-native';
+
+import type { Slots, UseSlots } from '@fluentui-react-native/framework';
+import { compose, mergeProps, withSlots } from '@fluentui-react-native/framework';
+import { IconV1 as Icon } from '@fluentui-react-native/icon';
+import { TextV1 as Text } from '@fluentui-react-native/text';
+
+import { useMenuItemRadio } from './useMenuItemRadio';
+import { getAccessibilityLabel, getTooltip } from '../MenuItem/MenuItem';
+import { stylingSettings } from '../MenuItemRadio/MenuItemRadio.styling';
+import type {
+  MenuItemRadioProps,
+  MenuItemRadioSlotProps,
+  MenuItemRadioInfo,
+  MenuItemRadioType,
+} from '../MenuItemRadio/MenuItemRadio.types';
+
+export const menuItemRadioName = 'MenuItemRadio';
+
+export const MenuItemRadio = compose<MenuItemRadioType>({
+  displayName: menuItemRadioName,
+  ...stylingSettings,
+  slots: {
+    root: Pressable,
+    radioInnerCircle: View,
+    radioButton: Pressable,
+    content: Text,
+    iconPlaceholder: View,
+    imgIcon: Image,
+    fontOrSvgIcon: Icon,
+  },
+  useRender: (userProps: MenuItemRadioProps, useSlots: UseSlots<MenuItemRadioType>) => {
+    const menuItem = useMenuItemRadio(userProps);
+    const Slots = useSlots(userProps, (layer): boolean => menuItem.state[layer] || userProps[layer]);
+    return menuItemRadioFinalRender(menuItem, Slots);
+  },
+});
+
+export const menuItemRadioFinalRender = (
+  menuItem: MenuItemRadioInfo,
+  Slots: Slots<MenuItemRadioSlotProps>,
+): React.FunctionComponent<MenuItemRadioProps> => {
+  return (final: MenuItemRadioProps, children: React.ReactNode) => {
+    const { accessibilityLabel, icon, tooltip, ...mergedProps } = mergeProps(menuItem.props, final);
+
+    const label = getAccessibilityLabel(accessibilityLabel, children[0]);
+    const tooltipResult = getTooltip(tooltip, menuItem.state.hasTooltips, children[0]);
+
+    return (
+      <Slots.root {...mergedProps} accessibilityLabel={label}>
+        <Slots.radioButton onPress={mergedProps.onPress} accessible={false} focusable={false}>
```

Comment:
I feel like in theory this should be something that you can replace the checkmark slot with from MenuItemCheckmark.. (i.e. use MenuItemCheckmark.compose to create the MenuItemRadio)

More of a one off comment than an actual suggestion.

Reply:
Do you mean MenuItemCheckbox 🤔  ? 
I think the 
MenuItemRadio.tsx file does it but it uses the Checkbox types and uses a function : menuItemFinalRender which render the jsx. 


First approach was that only but as comment to separate out radio from checkbox to created a seperate MenuItemRadio.android.tsx which renders the jsx for radio. 
I think styling would be issues if I compose MenuItemCheckbox , we wanted to have users to have direct access to tokens of Radio so that they can access it using customize like: 
MenuItemRadio.customize({ radioProperties: '...."  })

Reply:
Yeah that's a fair point. Makes sense to have to reimplement it then.